src/main/java/org/codehaus/mojo/gwt/AbstractGwtMojo.java

```diff
@@ -293,6 +321,17 @@ protected boolean isGenerator() {
         return files;
     }
 
+    private List<Exclusion> getProjectPluginExclusions() {
+      final List<Exclusion> exclusions = new ArrayList<Exclusion>();
+      final Plugin plugin = project.getPlugin(Plugin.constructKey(GWT_PLUGIN_GROUP_ID, GWT_PLUGIN_ARTIFACT_ID));
+      for ( final Dependency dep : plugin.getDependencies() )
+      {
+        exclusions.addAll( dep.getExclusions() );
```

Comment:
This is getting just the dependencies of the gwt-maven plugin and collects the exclusions so we don't add them to the classpath later on. Agreed, on the implementation not being pretty, but we couldn't find a better API to do it. Any advice would be much appreciated.